### DISCUSSION TITLE
NVM Node version is not recognised in neovide

### DISCUSSION THREAD

**Discussion by ckangnz** [neovide/neovide#2267] [General]
**Describe the bug**

My default node version is set to v14 (which i need for my project), but Neovide is defaulting to v18 which I don't have installed.
When opening Nvim, it works fine. Also it works fine on vim.

**To Reproduce**
1. Use `nvm use` in a project that has `.nvmrc` with specific version
2. Open neovide
3. Check the version

**Expected behavior**

I expect the node version used in the Neovide to match the node version I was using in the terminal

**Screenshots**
<img width="357" alt="image" src="https://user-images.githubusercontent.com/17423698/183639170-fc2aaf53-e019-4e6c-ad9f-83ae640ea092.png">

<img width="1143" alt="image" src="https://user-images.githubusercontent.com/17423698/183639356-9e3d9045-81e1-4e83-aaf8-dad24ea45851.png">

**Desktop (please complete the following information):**
 - OS: [e.g. macOS 12.5]
 - Neovide Version [e.g. 0.9.0]
 - Neovim Version [e.g. 0.7.2]

**Comment by MultisampledNight** [comment]
Thanks for reporting! If you use nvm according to https://github.com/nvm-sh/nvm#zsh, be aware that Neovide is not starting the embedded neovim instance not inside of a login shell, and as such zsh will _not_ source `~/.zshrc`. You could use `~/.zprofile` instead. I think I'll add this to the website.

**Reply by phkorn** [reply]
Is this already somewhere documented?
It took me a while to find this entry on how to set env variables for neovide.
I needed it for setting `TERM=xterm-256color`

**Reply by fredizzimo** [reply]
You probably don't need to set it, but there's a bug in Neovide that causes the built in terminal to not use true colors:
* https://github.com/neovide/neovide/issues/2288
* https://github.com/neovide/neovide/discussions/2252

But if you do need it, you can either set in in `.zprofile` or `.zshenv`.  We do have some documentation, but it needs to be expanded to cover other terminals and explain the situation a bit better. https://neovide.dev/faq.html#neovide-is-not-picking-up-some-shell-configured-information

These are some useful reference flows, but unfortunately for example the fish shell is mssing. Neovide uses a non-interactive login shell, that's the orange arrow.
![image](https://github.com/neovide/neovide/assets/9946255/0cd00ef6-ca31-42ba-a710-00278feb62fe)

**Comment by ckangnz** [comment]
I am, in fact, using "Calling nvm use automatically in a directory with a .nvmrc file" code in my `.zshrc`

I tried moving the snippet to `~/.zprofile` but it complains that it can't find `add-zsh-hook` in `.zprofile` because I believe the plugin has not loaded yet. (I also use zplug managing my plugins, which is loaded at the beginning of my .zshrc)

Is there an option for Neovide to source ~/.zshrc via a flag?

**Comment by cserb** [comment]
Just adding to the discussion here:
This seems to be a general issue that applies to multiple virtual dev environments (ruby, python, node). Currently I face the same issue with python which has a virtual env for every project and activating it is important for LSP. If I program in Python I have to go back to terminal nvim...
Having a solution rather than a workaround with e.g. `.zprofile` would be much appreciated

**Comment by ckangnz** [comment]
Has there been any update on this yet?

**Comment by MultisampledNight** [comment]
#1758 was merged which might be worth a look, you could build from git main and test. Doesn't fix the situation with Neovide being launched from Finder though.

**Comment by fredizzimo** [comment]
This works if you have your shell startup files correctly setup, for `zsh`,  using `.zprofile` and `.zshenv`, or if you configure the lsp to work without the tools in the path.

It's impossible to give a generic advice though, since every configuration is different, with different needs.

Neovide has to run as a noninteractive login shell, so `.zshrc` is not souced. Doing that would cause a lot of issues, since Neovide captures the stdout, and tools like `neofetch` and even zsh prompts moght infer with that. Also, launching an interactive shell without a terminal is against the intention of the shell type.
